<?xml version="1.0"?>
<robot name="flexible">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>

<link name="link1">
  <visual>
    <geometry>
      <box size="1 .2 .1"/>
    </geometry>
    <origin rpy="0 0 0" xyz=".5 0 0"/>
    <material name="white">
      <color rgba="1 1 1 1"/>
    </material>
  </visual>
</link>

<joint name="base_to_link1" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="base_link"/>
  <child link="link1"/>
  <origin xyz="0 0 .1"/>
</joint>

<link name="link2">
  <visual>
    <geometry>
      <box size="1 .2 .1"/>
    </geometry>
    <origin rpy="0 0 0" xyz=".5 0 0"/>
    <material name="blue">
      <color rgba="0 0 .8 1"/>
    </material>
  </visual>
</link>

<joint name="link1_to_link2" type="continuous">
  <axis xyz="0 0 1"/>
  <parent link="link1"/>
  <child link="link2"/>
  <origin xyz="1 0 0"/>
</joint>

<link name="end">
  <visual>
    <geometry>
      <box size="0 0 0"/>
    </geometry>
    <origin rpy="0 0 0" xyz=".5 0 0"/>
  </visual>
</link>

<joint name="link2_to_end" type="fixed">
  <parent link="link2"/>
  <child link="end"/>
  <origin xyz="1 0 0"/>
</joint>

</robot>
